R/backport_linter.R

```diff
@@ -7,12 +7,13 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 backport_linter <- function(r_version = getRversion()) {
+  if (inherits(r_version, "numeric_version")) r_version <- format(r_version)
```

Comment:
shouldn't we be doing the opposite? instead of comparing strings. it hasn't happened for any R versions but in principle `4.10.0` would break the logic here.

(doesn't need to be addressed in this PR)

Reply:
It's used later vs. `names(backports)` and contains non-version strings ("devel").
Not sure if that would break. Swapping the format and comparison would fix half of the problem at no cost, but I just noticed we need `4.1.0` backports in the data anyway, so lets create a follow-up.

Reply:
Fixed by #973